File: neatformcore/src/main/java/com/opensrp/neatformcore/domain/view/FormBuilder.java
package com.opensrp.neatformcore.domain.view;

import android.content.Context;
import android.view.ViewGroup;

import com.opensrp.neatformcore.domain.model.NForm;
import com.opensrp.neatformcore.views.data.ViewDataHandler;

public interface FormBuilder {
    void setMainLayout(ViewGroup mainLayout);

    ViewGroup getMainLayout();

    NForm getForm(String source);

    void addFormViews(NForm form, Context context);

    void setViewDataHandler(ViewDataHandler viewDataHandler);
}

File: neatformcore/src/main/java/com/nerdstone/neatformcore/domain/builders/FormBuilder.kt
package com.nerdstone.neatformcore.domain.builders

import android.content.Context
import android.view.View
import android.view.ViewGroup
import com.nerdstone.neatformcore.rules.RulesFactory

interface FormBuilder {

    var fileSource: String

    var mainLayout: ViewGroup

    fun buildForm(viewList: List<View>? = null): FormBuilder

    fun createFormViews(context: Context, views: List<View>? = null)

    fun registerFormRules(context: Context, rulesFileType: RulesFactory.RulesFileType)
}